### PR TITLE
Print message on `sinol-make` crash

### LINKED ISSUE #78: On crash show message where to submit issues

### PR DESCRIPTION
**Pull request by MasloMaslane**
Closes #78

### CODE REVIEW

**Comment by tonowak** [comment]
How does an exception now looks like for the user?

**Comment by tonowak** [comment]
Yeah, we don't really want to hide the reason from the user. I'm asking because the output format affects how the user might perceive the custom error message. Soon I'll try to improve it.

File: src/sinol_make/__init__.py [hunk 1]
```diff
@@ -56,7 +59,16 @@ def main():
...
     parser.print_help()
+
+
+def main():
+    try:
+        _main()
+    except:
+        print(traceback.format_exc())
+        util.exit_with_error('An error occurred while running the command.\n'
+                             'You can submit an issue here: https://github.com/sio2project/sinol-make/issues')
```
Comment (tonowak) [inline src/sinol_make/__init__.py:74]:
```suggestion
        util.exit_with_error('An error occurred while running the command.\n'
                             'If that is a bug, please report it or submit a bugfix: https://github.com/sio2project/sinol-make/#reporting-bugs-and-contributing-code')
```

File: src/sinol_make/__init__.py [hunk 2]
```diff
@@ -30,7 +32,7 @@ def configure_parsers():
     return parser
 
 
-def main():
+def _main():
```
Comment (tonowak) [inline src/sinol_make/__init__.py:35]:
```suggestion
def main_exn():
```

Is there any Python conventions to indicate that a function might throw? I'm familiar with the `_exn` suffix in other languages, but I'm not sure how it is in Python.

Comment (MasloMaslane) [reply src/sinol_make/__init__.py:35]: I didn’t find any, but I think we should add it here

File: src/sinol_make/__init__.py [hunk 3]
```diff
@@ -30,7 +32,7 @@ def configure_parsers():
     return parser
 
 
-def main():
+def _main_exn():
```
Comment (tonowak) [inline src/sinol_make/__init__.py:35]:
```suggestion
def main_exn():
```
I don't see any reason to hide this function now that its name describes what it does.

**Review by tonowak — APPROVED** [review]
You can merge after resolving this comment

**Pull request merged**